idflakies-core/src/main/java/edu/illinois/cs/dt/tools/utility/Pair.java

```diff
@@ -0,0 +1,58 @@
+package edu.illinois.cs.dt.tools.utility;
+
+import java.io.Serializable;
+
+
+/**
+ * A (key, value) pair.
+ */
+public class Pair<K, V> implements Serializable {
+    private K key;
+    private V value;
+
+    public Pair(K key, V value) {
+        this.key = key;
+        this.value = value;
+    }
+
+    @Override
+    public String toString() {
+        return key + "," + value;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        Pair<?, ?> pair = (Pair<?, ?>) obj;
+
+        if (!key.equals(pair.key)) {
```

Comment:
Minor thing, this can actually just be simplified into `return key.equals(pair.key) && value.equals(pair.value)`